packages/realm-react/src/useObject.tsx

```diff
@@ -21,18 +21,30 @@ import Realm from "realm";
 
 import { CachedObject, createCachedObject } from "./cachedObject";
 import { CollectionCallback, getObjectForPrimaryKey, getObjects } from "./helpers";
+import { UseRealmHook } from "./useRealm";
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+type AnyRealmObject = Realm.Object<any>;
+
+export type UseObjectHook = {
+  <T>(type: string, primaryKey: T[keyof T], keyPaths?: string | string[]): (T & Realm.Object<T>) | null;
+  <T extends AnyRealmObject>(
+    type: { new (...args: any): T },
+    primaryKey: T[keyof T],
+    keyPaths?: string | string[],
+  ): T | null;
+};
```

Comment:
Since we're now deprecating the usage of separate arguments for `useQuery()` in favor of an options arg, it introduces some inconsistency between that API and that of `useObject()`. Even though `useObject()` doesn't have the ESLint issue regarding dependency lists, I think we should consider allowing an options argument for that hook as well, especially since the third arg `keyPaths` has been introduced.

I'd be interested in hearing your thoughts on this @kraenhansen @takameyer .

Reply:
Good consideration.  I would be in favor of using an options object for `useObject()`.

Reply:
True -- that would probably be more consistent and read better for some users 👍

Reply:
I've push an update including that 👍